### PR TITLE
fix(AlertReports): preventing reset of notification settings when method does not change

### PR DESCRIPTION
**Pull request by fisjac**
### SUMMARY

When selecting a method for notification settings in the alerts and reports modal, the notification settings are deleted when the same method is selected. This PR addresses this issue by preventing an update to the notificationSettings when no the method value has not changed.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Before:
https://www.loom.com/share/9f9829e8877a444db074e6f8ce0f669f?sid=1e4fe88d-4075-4b6a-9dc0-50749316baa2

After:
https://www.loom.com/share/906d409dfbd14d56ab2e4e0ee0f1c799?sid=8464d755-954b-4787-a9d1-4942258b2f49

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Required feature flags: ALERT_REPORTS
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/features/alerts/components/NotificationMethod.tsx [hunk 1]
```diff
@@ -76,6 +76,9 @@ export const NotificationMethod: FunctionComponent<NotificationMethodProps> = ({
   }
 
   const onMethodChange = (method: NotificationMethodOption) => {
```
Comment (john-bodley) [inline superset-frontend/src/features/alerts/components/NotificationMethod.tsx:78]: Ideally the `onMethodChange` shouldn't fire if the value isn't changed. It's likely worthwhile checking that the `Select` component logic is correct.

Comment (fisjac) [reply superset-frontend/src/features/alerts/components/NotificationMethod.tsx:78]: Opening a new issue https://github.com/apache/superset/issues/27668

**Comment by fisjac** [comment]
Closing in favor of another PR to address upstream issue https://github.com/apache/superset/issues/27668

**Comment by fisjac** [comment]
Resolved in the following PR:
https://github.com/apache/superset/issues/27668

**Pull request closed without merging**